x/dex/keeper/grpc_query_get_twaps.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	"context"
+	"sort"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/sei-protocol/sei-chain/x/dex/types"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+)
+
+func (k Keeper) GetTwaps(goCtx context.Context, req *types.QueryGetTwapsRequest) (*types.QueryGetTwapsResponse, error) {
+	if req == nil {
+		return nil, status.Error(codes.InvalidArgument, "invalid request")
+	}
+
+	ctx := sdk.UnwrapSDKContext(goCtx)
+	allRegisteredPairs := k.GetAllRegisteredPairs(ctx, req.ContractAddr)
+	twaps := []*types.Twap{}
+	for _, pair := range allRegisteredPairs {
+		prices := k.GetAllPrices(ctx, req.ContractAddr, pair)
+		twaps = append(twaps, &types.Twap{
+			Pair:            &pair,
+			Twap:            calculateTwap(ctx, prices, req.LookbackSeconds),
+			LookbackSeconds: req.LookbackSeconds,
+		})
+	}
+
+	return &types.QueryGetTwapsResponse{
+		Twaps: twaps,
+	}, nil
+}
+
+func calculateTwap(ctx sdk.Context, prices []*types.Price, lookback uint64) sdk.Dec {
+	// sort prices in descending order to start iteration from the latest
+	sort.Slice(prices, func(p1, p2 int) bool {
```

Comment:
since the ordering won't change, we could consider using a priority queue to reduce number of sorts . If number of prices is pretty small then probably doesn't matter